redirector/common.py

```diff
@@ -83,14 +83,14 @@ def get_ddb_table():
 def get_file_pathname(file):
     # want to start the path after "All Files/<BoxFolderName>/"
     file_path_collection = file.path_collection
-    file_path_names = [fp.get().name for fp in file_path_collection["entries"][2:]]
-    if len(file_path_names) > 0:
-        file_path = "/".join(file_path_names)
-        fullpath = "/".join([file_path, file.name])
-    else:
-        fullpath = file.name
-
-    return fullpath
+    start_index = [e.id for e in file_path_collection["entries"]].index(BOX_FOLDER_ID) + 1
+    file_path_names = [fp.get().name for fp in file_path_collection["entries"][start_index:]] + [file.name]
+    # note: lists with len < 1 are not handled. They shouldn't happen.
```

Comment:
I think the elif branch here returns a list (`["filename.ext"]`) and not a string.  What do you think of this instead of the if/else block:

```python
return "/".join(file_path_names)
```

if `file_path_names = ["filename.ext"]`, that evaluates to `"filename.ext"`
if `file_path_names = ["some", "path", "to", "filename.ext"]`, that evaluates to `"some/path/to/filename.ext"`

I think it covers all the cases?

Reply:
the issue with `"/".join()` is that if the only item is the file (i.e. it's not nested at all), then the leading slash makes it into the db, but the redirector queries the table without a leading slash.

Reply:
Deleted previous comment. Somehow yesterday I was getting leading slashes into the DynamoDB, and I thought it was the join.